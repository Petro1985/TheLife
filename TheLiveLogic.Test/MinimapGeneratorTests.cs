using System.Collections.Generic;
using System.Collections.Immutable;
using System.Drawing;
using Snapshooter.Xunit;
using TheLifeServices.Services;
using TheLiveLogic.Fields;
using Xunit;

namespace TheLiveLogic.Test;

public class MinimapGeneratorTests
{
    private readonly MinimapGenerator _generator;

    public MinimapGeneratorTests()
    {
        _generator = new MinimapGenerator();
    }

    [Fact]
    public void MinimapGenerator_Should_Generate_Minimap()
    {
        var map = new List<Coord>()
        {
            new Coord(3, 3),
            new Coord(4, 4),
            new Coord(4, 5),
            new Coord(3, 5),
            new Coord(2, 5),
            new Coord(-5, -5),
            new Coord(-4, -4),
            new Coord(-6, -6),
            new Coord(-3, -3)
        };

        var img = _generator.Generate(map, 140);

        var qwe = new ImageConverter();
        var stringImage = (byte[])qwe.ConvertTo(img, typeof(byte[]));

        Snapshot.Match(stringImage);
    }
}