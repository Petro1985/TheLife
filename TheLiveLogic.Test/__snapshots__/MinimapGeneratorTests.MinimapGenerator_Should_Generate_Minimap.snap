"iVBORw0KGgoAAAANSUhEUgAAAIwAAACMCAYAAACuwEE+AAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAAJ1SURBVHhe7dJBakQxDETBuXnm5pMLtAUNDh7/1KI2BguB3uvn8/o03u83/5hgqAiGimCoCIaKYKgIhopgqAiGimCoCIaKYKi80uMkRbSS/nM3wVARDBXBUBEMFcFQEQwVwVARDBXBUBEMlTqYRopokmbwXQRDRTBUBENFMFQEQ0UwVARDRTBUBENFMFQEQ+VPg2mliCZpBn9LMFQEQ0UwVARDRTBUBENFMFQEQ0UwVARDRTBUvioYvp9gqAiGimCoCIaKYKgIhopgqAiGimCoCIaKYKgIhopgqAiGimCoCIaKYKgIhopgqAiGimCoCIaKYKgIhopgqAiGimCoCIaKYKgIhopgqAiGimCoCIaKYKgIhopgqAiGimCoCIaKYKgIhopgqAiGimCoCIaKYKgIhopgqAiGimCoCIaKYKgIhopgqAiGimCoCOYCP59XJc3YRTAXSFFM0oxdBHOBFMUkzdhFMBdIUUzSjF0Ec4EUxSTN2EUwF0hRTNKMXQRzgRTFJM3YRTAXSFFM0oxdBHOBFMUkzdhFMBdIUUzSjF0Ec4EUxSTN2EUwF0hRTNKMXQTzQCmilfR/IpgHSmGspP8TwTxQCmMl/Z8I5oFSGCvp/0QwD5TCWEn/J4J5oBTGSvo/EcwDpTBW0v+JYB4ohbGS/k8E80ApjJX0fyKYB0phrKT/E8E8UApjJf2fCOaAdLhT0n4TwRyQDndK2m8imAPS4U5J+00Ec0A63Clpv4lgDkiHOyXtNxHMAelwp6T9JoI5IB3ulLTfRDAHpMOdkvabCOaAdLhT0n4TwRyQDndK2m8imAPS4U5J+00Ec0A63Clpv7X35xf8uhz2LD+nQwAAAABJRU5ErkJggg=="
